### ISSUE TITLE
Ticket was not reopened by customer web channel follow up reply with enabled group follow up settings

### ISSUE THREAD

**Issue by antonionardella** [zammad/zammad#5332]
### Used Zammad Version

6.3.1-1724757244.41b98ccb.noble

### Environment

- Installation method: package
- Operating system (if you're unsure: `cat /etc/os-release` ): ubuntu 24.04
- Database + version: postgresql 16.4
- Elasticsearch version: 7.10.2
- Browser + version: Firefox 129.0.2 (64-bit)

### Actual behaviour

On a closed ticket in a group with follow up configurations
The customer replied to the ticket to a closed ticket in the web channel, but the ticket status remained “closed successful” (custom state of type closed)
![2c2a728618f218b6a10103f5fbbc1557de18d304](https://github.com/user-attachments/assets/9d43c9e6-9dc2-49f2-a011-e360bfd6022d)

### Expected behaviour

The customer replied to a closed ticket via web, the status should change back to open given the follow-up settings of the group the ticket resides in

### Steps to reproduce the behaviour

Set the group settings as follows
![633ab28b18412bab24a36d27d17b2a6cb626b7ca](https://github.com/user-attachments/assets/f0646024-975c-4504-9980-8349ca31bf30)
As agent set to “closed successful” (custom state of type closed as explained above)
Customer replies to closed ticket via web channel
Ticket remains closed

Ticket history
```
08/19/2024
Customer
    created Ticket
    created Article
    updated Ticket Last contact '08/19/2024 9:39 am'
    updated Ticket Last contact (customer) '08/19/2024 9:39 am'
    updated Ticket Escalation at (First Response Time) '08/21/2024 12:09 pm'
    updated Ticket Escalation at '08/21/2024 12:09 pm'

Trigger: auto reply (on new tickets)

    created Article

08/19/2024
System

    email sent to 'cuseromer@email.com'

    notification Ticket 'agent@company.com(create:online,email)'
    notification Ticket 'agent@company.com(create:online,email)'
    notification Ticket 'owneragent@company.com(create:online,email)'
    notification Ticket 'agent@company.com(create:online,email)'

08/19/2024
Owner Agent
    created Article

08/19/2024
Owner Agent
    updated Ticket Owner from '-' → 'Owner Agent'
    updated Ticket last_owner_update_at '08/19/2024 9:49 am'
    updated Ticket Escalation at (First Response Time) '08/21/2024 12:09 pm' → '-'
    updated Ticket Escalation at '08/21/2024 12:09 pm' → '-'

08/19/2024
Owner Agent
    updated Ticket State from 'new' → 'pending reminder'
    updated Ticket last_owner_update_at '08/19/2024 9:49 am' → '-'
    updated Ticket Pending till '08/19/2024 5:00 pm'

08/19/2024
System

    notification Ticket 'owneragent@company.com(reminder_reached:online,email)'

08/20/2024
System

    notification Ticket 'owneragent@company.com(reminder_reached:online,email)'

08/21/2024
System

    notification Ticket 'owneragent@company.com(reminder_reached:online,email)'

08/22/2024
System

    notification Ticket 'owneragent@company.com(reminder_reached:online,email)'

08/23/2024
System

    notification Ticket 'owneragent@company.com(reminder_reached:online,email)'

08/23/2024
Owner Agent
    updated Ticket State from 'pending reminder' → 'closed successful'
    updated Ticket Closing time '08/23/2024 11:40 am'
    updated Ticket Last closing time '08/23/2024 11:40 am'
    updated Ticket Pending till '08/19/2024 5:00 pm' → '-'
    created Article

19 hours ago
Customer
    created Article
    notification Ticket 'owneragent@company.com(update:online,email)'

17 hours ago
Owner Agent
    updated Ticket State from 'closed successful' → 'open'
    updated Ticket last_owner_update_at '09/03/2024 3:49 pm'

11 minutes ago
Owner Agent
    updated Article internal from 'true' → 'false'
    updated Article internal from 'true' → 'false'
```

Thank you

### I'm sure this is a bug and no feature request or a general question.

yes

**Comment by dominikklein** [comment]
@antonionardella But in your example group configuration it's configured that it should not be "reopen".

**Comment by dominikklein** [comment]
@antonionardella Oki tested it again. In my test with such a group situation, a follow-up is not possible. Oki, you are seeing the text field, but when you try to submit an error is coming from the server: `Cannot follow-up on a closed ticket. Please create a new ticket.`

When follow-up is possible the form automatically changes the closed state to open (but the user can manually change this value back to "closed").

**Comment by antonionardella** [comment]
Hello @dominikklein,

Thanks for taking the time and sorry, as I am confused by your feedback.

The group is set to `do not reopen ticket after certain time but create new ticket`, and the `Reopening time in days` is set to `14`.
According to the tooltip `Allow reopening of tickets wihin a certain time` and the time is 14 days.
![image](https://github.com/user-attachments/assets/a44f31eb-ddf8-44b1-a1e1-7989f61ca79e)

And the group setting according to the [official documentation](https://admin-docs.zammad.org/en/latest/manage/groups/settings.html):

>**The ticket can be reopened unless the specified number of days after last ticket closure has been exceeded**. If the time limit is exceeded, Zammad will create a new ticket instead of reopening.
>Choosing this option will provide the option Reopening time in days which requires you to provide the number of days you want Zammad to wait until it creates new tickets.
![image](https://github.com/user-attachments/assets/58584c9e-ebfc-444a-83c3-9f9658ec7743)

Which I understand in our case as "A closed ticket will be reopened if the time between the closing status and now is less than 14 days".

Please let me know if I understand the configuration wrong.

**Comment by dominikklein** [comment]
Correctly, when the customer sends an mail or use the web interface in this range, it's possible to add an note.

**Comment by antonionardella** [comment]
Thank you for confirming that the configuration is correct.

Now back to the issue:
According to the history the agent closed the ticket
```
08/23/2024
Owner Agent
    updated Ticket State from 'pending reminder' → 'closed successful'
    updated Ticket Closing time '08/23/2024 11:40 am'
    updated Ticket Last closing time '08/23/2024 11:40 am'
    updated Ticket Pending till '08/19/2024 5:00 pm' → '-'
    created Article
```

Then the customer replied to the closed ticket
```
19 hours ago
Customer
    created Article
    notification Ticket 'owneragent@company.com(update:online,email)'
```

The is no record of the system reopening the ticket, but the agent had to reopen it itself
```
17 hours ago
Owner Agent
    updated Ticket State from 'closed successful' → 'open'
    updated Ticket last_owner_update_at '09/03/2024 3:49 pm'

11 minutes ago
Owner Agent
    updated Article internal from 'true' → 'false'
    updated Article internal from 'true' → 'false'
```

I got at least other 2 times this happened according to the agent and I tried it on my staging system and I can confirm that the system does not reopen a ticket with this configuration if the customer adds a note using the web channel.

I hope this helps to figure out if I am doing something wrong or if there really is a bug here.

Best,
Antonio

**Comment by dominikklein** [comment]
At least I can not reproduce the problem. Like I said normally the form is switching the state automatically, which is working for me with all different configurations. However, the customer can switch the state back to "closed" before submitting.

Feel free to check with a customer user that you can reproduce the mentioned problem and maybe add some more details, but for now, I would close the issue.

**Comment by antonionardella** [comment]
Thank you for taking the time. I see now where the misunderstanding of this functionality lies within.

The status change is not something the _system_ does because there is an incoming message in the web channel, but if the customer starts typing a note within the browser, the status is _automatically_ changed to open in the top right.

![image](https://github.com/user-attachments/assets/81ea85e0-c44b-4187-be2f-e8d302a26236)

So, if the administrator sets up a core workflow that denies a customer the possibility to change status from closed to open, the status change will not happen _automatically_ if the web channel is used.

In short if an email is sent, the system receives an email, the status is changed by the system.
If the web channel is used by the customer, the customer must have permissions to change states.

Thank you again for your time, let me close the issue.
